### PR TITLE
[AIRFLOW-1919] Add option to query for DAG runs given a DAG ID

### PR DESCRIPTION
**Pull request by feng-tao**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1919
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
 Add a cli option to list_dag_runs for given dag id. It support filter only certain states, filter out backfill dagruns.

Here is the sample output:
------------------------------------------------------------------------------------------------------------------------
DAG RUNS
------------------------------------------------------------------------------------------------------------------------
id  | run_id               | state      | execution_date       | state_date           |

3   | backfill_2018-06-04T00:00:00+00:00 | success    | 2018-06-04T00:00:00+00:00 | 2018-06-16T06:06:43.716135+00:00 |
4   | backfill_2018-06-02T00:00:00+00:00 | success    | 2018-06-02T00:00:00+00:00 | 2018-06-16T06:12:50.650013+00:00 |
1   | backfill_2018-06-01T00:00:00+00:00 | success    | 2018-06-01T00:00:00+00:00 | 2018-06-16T05:51:21.070803+00:00 |

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Unit test added

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by feng-tao** [comment]
PTAL @Fokko @kaxil

File: tests/core.py [hunk 1]
```diff
@@ -1004,6 +1004,14 @@ def test_cli_list_dags(self):
         args = self.parser.parse_args(['list_dags', '--report'])
         cli.list_dags(args)
 
+    def test_cli_list_dag_runs(self):
+        cli.trigger_dag(self.parser.parse_args([
+            'trigger_dag', 'example_bash_operator', ]))
+        args = self.parser.parse_args(['list_dag_runs',
+                                       'example_bash_operator',
+                                       '--no_backfill'])
+        cli.list_dag_runs(args)
```
Comment (Fokko) [inline tests/core.py:1013]: We don't test anything here, right? :)

Comment (feng-tao) [reply tests/core.py:1013]: hey @Fokko , I think all the cli tests in airflow(CliTest) are tested in this way which make sure the cli is not thrown any exceptions.

**Comment by feng-tao** [comment]
hey @Fokko , do you have any other concern on the pr?

**Comment by feng-tao** [comment]
PTAL @Fokko @kaxil

**Review by Fokko — APPROVED** [review]
LGTM

**Pull request closed without merging**